Src/MoneyFox.Core/ApplicationCore/Queries/GetPaymentsForAccountIdQuery/GetPaymentsForAccountIdQuery.cs

```diff
@@ -20,15 +20,22 @@ public class GetPaymentsForAccountIdQuery : IRequest<List<Payment>>
             DateTime timeRangeEnd,
             bool isClearedFilterActive = false,
             bool isRecurringFilterActive = false,
-            PaymentTypeFilter filteredPaymentType = PaymentTypeFilter.All)
+            PaymentTypeFilter filteredPaymentType = PaymentTypeFilter.All,
+            string sortCriterion = "date",
```

Comment:
The sorting of the grid is in my opinion only something that regards the WinUI view. Therefore I think the sort direction and criterion should not leak into the core but should be handled in the ViewModel of the PaymentList.
Instead of loading everytime all the data again with a different sorting you could just load them once and reorder in the viewmodel on the loaded list.